src/language-js/print/binaryish.js

```diff
@@ -267,8 +278,11 @@ function printBinaryishExpressions(
           "right"
         )
       : print("right");
+    // Place leading own line non-typecast comments before the operator.
+    const comment = commentBeforeOperator ? rightContent.splice(0, 1)[0] : "";
```

Comment:
note: this will crash with code like
```ts
foo
// comment
|| foo.method().method();
```

The `foo.method().method()` creates a "label" object.